pkg/validations/upgradevalidations/cluster.go

```diff
@@ -26,11 +26,25 @@ func ValidateClusterObjectExists(ctx context.Context, k validations.KubectlClien
 	return fmt.Errorf("couldn't find CAPI cluster object for cluster with name %s", cluster.Name)
 }
 
-func ValidateTaintsSupport(ctx context.Context, clusterSpec *cluster.Spec) error {
+func ValidateTaintsSupport(clusterSpec *cluster.Spec) error {
 	if !features.IsActive(features.TaintsSupport()) {
 		if len(clusterSpec.Cluster.Spec.ControlPlaneConfiguration.Taints) > 0 {
 			return fmt.Errorf("Taints feature is not enabled.")
 		}
 	}
 	return nil
 }
+
+func ValidateNodeLabelsSupport(clusterSpec *cluster.Spec) error {
```

Comment:
Do we need the same code duplicated in two places?
Maybe we could move it to a common package and share it

Reply:
I just kept it the same as taints for now because i don't want to introduce too many changes right now right before the release. But if we are okay with doing that I prefer keeping it in one place too